geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ClientUserAuths.java

```diff
@@ -121,7 +147,7 @@ public UserAuthAttributes getUserAuthAttributes(final String cqName) {
   public Subject getSubject(final String cqName) {
     Long uniqueId = cqNameVsUniqueId.get(cqName);
     if (uniqueId != null) {
-      return uniqueIdVsSubject.get(uniqueId);
+      return uniqueIdVsSubject.get(uniqueId).get(0);
```

Comment:
Need a null check before calling `get(0)`. Otherwise it could throw `NullPointerException`.

Reply:
good catch!